src/services/directionsManager.service.ts

```diff
@@ -425,26 +425,26 @@ export class DirectionsManagerService {
     //move outside first so there's a clean start in the next building without any corruption of data
     if(!this.router.url.includes('outdoor')) {
       this.router.navigateByUrl('/outdoor');
-    } 
-    
-    //move to the hall buildin
-    if (this.currentStep.floor.includes('h')) {
-      if (this.currentStep.floor === 'h8') {
-        this.router.navigateByUrl('/indoor/hall');
-      } else {
-        this.router.navigateByUrl('/indoor/hall');
-        //set timeout to allow all components to be loaded
-        setTimeout(() => {
-          this.changeFloor(this.currentStep.floor);
-        }, 1000);
-      }
```

Comment:
SetTimeout is requested here for the calendar, if a problem is the component is never loaded, imagine a user goes outside, go to his calendar click direction to is 900 class, get outside, click button to go inside the component never had time to subscribe to everything so it goes to 8th floor